File: backend/src/main/java/com/nexustasks/category/entity/Category.java
package com.nexustasks.category.entity;

import com.nexustasks.common.entity.BaseEntity;
import com.nexustasks.user.entity.User;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "categories")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Category extends BaseEntity {

    @Column(nullable = false, length = 100)
    private String name;

    @Column(nullable = false, length = 100)
    private String slug;

    @Column(length = 500)
    private String description;

    @Column(nullable = false, length = 7)
    @Builder.Default
    private String color = "#6b7280";

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "owner_id", nullable = false)
    private User owner;
}

File: mobile/lib/features/categories/domain/models/category.dart
class Category {
  const Category({
    required this.publicId,
    required this.name,
    required this.color,
    this.description,
  });

  final String publicId;
  final String name;
  final String color;
  final String? description;

  factory Category.fromJson(Map<String, dynamic> json) => Category(
    publicId: json['publicId'] as String,
    name: json['name'] as String,
    color: (json['color'] as String?) ?? '#6b7280',
    description: json['description'] as String?,
  );
}
